open_spiel/games/phantom_go_test.cc

```diff
@@ -0,0 +1,231 @@
+{ 
+    GameParameters params;
+    params["board_size"] = GameParameter(kBoardSize);
+    std::shared_ptr<const Game> game =
+        LoadGame("phantom_go", params);
+    PhantomGoState state(game, kBoardSize, kKomi, 0);
+    state.ApplyAction(5);
+
+    std::unique_ptr<State> stateClone = state.Clone();
+
+    SPIEL_CHECK_EQ(state.ToString(), stateClone->ToString());
+    SPIEL_CHECK_EQ(state.History(), stateClone->History());
+
+    state.ApplyAction(8);
+
+    SPIEL_CHECK_FALSE(state.ToString() == stateClone->ToString());
+    SPIEL_CHECK_FALSE(state.History() == stateClone->History());
+}
+
+void HandicapTest() {
+  std::shared_ptr<const Game> game =
+      LoadGame("phantom_go", {{"board_size", open_spiel::GameParameter(kBoardSize)},
+                      {"komi", open_spiel::GameParameter(kKomi)},
+                      {"handicap", open_spiel::GameParameter(1)}});
+  PhantomGoState state(game, kBoardSize, kKomi, 2);
+  SPIEL_CHECK_EQ(state.CurrentPlayer(), ColorToPlayer(GoColor::kWhite));
+  SPIEL_CHECK_EQ(state.board().PointColor(MakePoint("d4")), GoColor::kBlack);
+  
+}
+
+void IllegalMoveTest()
+{
+    GameParameters params;
+    params["board_size"] = GameParameter(kBoardSize);
+    std::shared_ptr<const Game> game =
+        LoadGame("phantom_go", params);
+    PhantomGoState state(game, kBoardSize, kKomi, 0);
+    SPIEL_CHECK_EQ(state.CurrentPlayer(), ColorToPlayer(GoColor::kBlack));
+    state.ApplyAction(5);
+    SPIEL_CHECK_EQ(state.CurrentPlayer(), ColorToPlayer(GoColor::kWhite));
+    state.ApplyAction(5);
+    SPIEL_CHECK_EQ(state.CurrentPlayer(), ColorToPlayer(GoColor::kWhite));
+}
+
+void StoneCountTest()
+{
+    GameParameters params;
+    params["board_size"] = GameParameter(kBoardSize);
+    std::shared_ptr<const Game> game =
+        LoadGame("phantom_go", params);
+    PhantomGoState state(game, kBoardSize, kKomi, 0);
+    SPIEL_CHECK_EQ(state.board().getStoneCount()[(uint8_t)GoColor::kBlack], 0);
+    SPIEL_CHECK_EQ(state.board().getStoneCount()[(uint8_t)GoColor::kWhite], 0);
+    state.ApplyAction(5);
+    SPIEL_CHECK_EQ(state.board().getStoneCount()[(uint8_t)GoColor::kBlack], 1);
+    SPIEL_CHECK_EQ(state.board().getStoneCount()[(uint8_t)GoColor::kWhite], 0);
+    state.ApplyAction(6);
+    SPIEL_CHECK_EQ(state.board().getStoneCount()[(uint8_t)GoColor::kBlack], 1);
+    SPIEL_CHECK_EQ(state.board().getStoneCount()[(uint8_t)GoColor::kWhite], 1);
+
+}
+
+void ConcreteActionsAreUsedInTheAPI() {
+  std::shared_ptr<const Game> game =
+      LoadGame("phantom_go", {{"board_size", open_spiel::GameParameter(kBoardSize)}});
+  std::unique_ptr<State> state = game->NewInitialState();
+
+  SPIEL_CHECK_EQ(state->NumDistinctActions(), kBoardSize * kBoardSize + 1);
+  SPIEL_CHECK_EQ(state->LegalActions().size(), state->NumDistinctActions());
+  for (Action action : state->LegalActions()) {
+    SPIEL_CHECK_GE(action, 0);
+    SPIEL_CHECK_LE(action, kBoardSize * kBoardSize);
+  }
+}
+
+void ResampleFromInfostateVisualTest()
```

Comment:
This "test" does not test anything -- there are no checks. Perhaps add a check the observation strings are identical?

Reply:
I am aware of the fact, that this is not the test that should be done. This is an implementation that I used during the development of the Metaposition Resampling and I still use it when implementing the Infostate Resampling. The reason that I used this visual test and not a classical test is that visually comparing resampled state and the observation is much more precise to find bugs than using a method to compare them. As soon as both the methods are finished, i will remove this method from the tests.